src/App.js

```diff
@@ -3,21 +3,31 @@ import TodoList from "./TodoList";
 import AddTodoForm from "./AddTodoForm";
 import { useState, useEffect } from "react";
 
-// /*This is the app that will display the arrays of Object and the form with button*
 
-const useSemiPersistentState = () => {
+
+function App() {
   const [todoList, setTodoList] = useState(
     JSON.parse(localStorage.getItem("savedTodoList")) || []
   );
 
+  const [isLoading, setIsLoading] = useState(true);
+
   useEffect(() => {
-    localStorage.setItem("savedTodoList", JSON.stringify(todoList));
-  }, [todoList]); 
-    return [todoList, setTodoList];
-};
+    new Promise((resolve, reject) => {
+      setTimeout(() => {
+        resolve({ data: { todoList: todoList } });
+      }, 2000);
+    }).then((result) => {
+      setTodoList(result.data.todoList);
+      setIsLoading(false);
+    });
+  });
```

Comment:
Make sure to include an empty dependency array in this `useEffect`
```suggestion
  }, []);
```